src/lib.rs

```diff
@@ -600,37 +600,33 @@ fn implement_plan<'a, 'b, A: Allocate, T: Timestamp+Lattice>(
         },
         &Plan::Lookup(e, a, sym1) => {
             let tuple = (vec![Value::Eid(e), Value::Attribute(a)], Default::default(), 1);
-            let ea_in = Some(tuple).to_stream(nested).as_collection();
+            let ea_in = Some(tuple).to_stream(nested).as_collection().arrange_by_self();
             let tuples = db.ea_v.enter(nested)
-                .semijoin(&ea_in)
-                .map(|x| x.1);
+                .join_core(&ea_in, |_,tuples,_| Some(tuples.clone()));
```

Comment:
Is there a reason for the change to `tuples` in plural here? Isn't that closure called on each resulting tuple?